crates/rmf_site_picking/src/select/resources.rs

```diff
@@ -2,13 +2,25 @@ use bevy_derive::{Deref, DerefMut};
 pub use bevy_ecs::prelude::*;
 use bytemuck::TransparentWrapper;
 use crossflow::Service;
+use std::collections::HashSet;
 
 use crate::*;
 use web_time::Instant;
 
 /// Used as a resource to keep track of which entity is currently selected.
-#[derive(Default, Debug, Clone, Copy, Deref, DerefMut, Resource)]
-pub struct Selection(pub Option<Entity>);
+#[derive(Default, Debug, Clone, Resource)]
+pub struct Selection {
+    pub selected: HashSet<Entity>,
+}
+
+impl Selection {
+    pub fn get_single(&self) -> Option<Entity> {
+        if self.selected.is_empty() || self.selected.len() > 1 {
```

Comment:
Small nitpick, you don't need to check for `self.selected.is_empty()` here because it will naturally come out as `None` for `self.selected.iter().next().copied()`.